internal/v2/s2av2_test.go

```diff
@@ -0,0 +1,146 @@
+	if _, _, err := sc.ServerHandshake(nil); err == nil {
+		t.Errorf("c.ServerHandshake(nil) should fail with incorrect transport credentials")
+	}
+}
+
+func TestInfo(t *testing.T) {
+	c, err := NewClientCreds("")
+	if err != nil {
+		t.Fatalf("NewClientCreds(\"\") failed: %v", err)
+	}
+	info := c.Info()
+	if got, want := info.SecurityProtocol, "s2av2"; got != want {
+		t.Errorf("info.SecurityProtocol=%v, want %v", got, want)
+	}
+}
+
+func TestCloneClient(t *testing.T) {
+	c, err := NewClientCreds("")
+	if err != nil {
+		t.Fatalf("NewClientCreds(\"\") failed: %v", err)
+	}
+	cc := c.Clone()
+	s2av2Creds, ok := c.(*s2av2TransportCreds)
+	if !ok {
+		t.Fatal("the created creds is not of type s2av2TransportCreds")
+	}
+	s2av2CloneCreds, ok := cc.(*s2av2TransportCreds)
+	if !ok {
+		t.Fatal("the created clone creds is not of type s2aTransportCreds")
+	}
+	if got, want := cmp.Equal(s2av2Creds, s2av2CloneCreds, protocmp.Transform(), cmp.AllowUnexported(s2av2TransportCreds{})), true; got != want {
+		t.Errorf("cmp.Equal(%v, %v) = %v, want %v", s2av2Creds, s2av2CloneCreds, got, want)
+	}
+	// Change the values and verify the creds were deep copied.
+	s2av2CloneCreds.info.SecurityProtocol = "s2a"
+	if got, want := cmp.Equal(s2av2Creds, s2av2CloneCreds, protocmp.Transform(), cmp.AllowUnexported(s2av2TransportCreds{})), false; got != want {
+		t.Errorf("cmp.Equal(%v, %v) = %v, want %v", s2av2Creds, s2av2CloneCreds, got, want)
+	}
+}
+
+func TestCloneServer(t *testing.T) {
+	c, err := NewServerCreds()
+	if err != nil {
+		t.Fatalf("NewServerCreds() failed: %v", err)
+	}
+	cc := c.Clone()
+	s2av2Creds, ok := c.(*s2av2TransportCreds)
+	if !ok {
+		t.Fatal("the created creds is not of type s2av2TransportCreds")
+	}
+	s2av2CloneCreds, ok := cc.(*s2av2TransportCreds)
+	if !ok {
+		t.Fatal("the created clone creds is not of type s2aTransportCreds")
+	}
+	if got, want := cmp.Equal(s2av2Creds, s2av2CloneCreds, protocmp.Transform(), cmp.AllowUnexported(s2av2TransportCreds{})), true; got != want {
+		t.Errorf("cmp.Equal(%v, %v) = %v, want %v", s2av2Creds, s2av2CloneCreds, got, want)
+	}
+	// Change the values and verify the creds were deep copied.
+	s2av2CloneCreds.info.SecurityProtocol = "s2a"
+	if got, want := cmp.Equal(s2av2Creds, s2av2CloneCreds, protocmp.Transform(), cmp.AllowUnexported(s2av2TransportCreds{})), false; got != want {
+		t.Errorf("cmp.Equal(%v, %v) = %v, want %v", s2av2Creds, s2av2CloneCreds, got, want)
+	}
+}
+
+func TestOverrideServerName(t *testing.T) {
+	wantServerName := "server.name"
+	c, err := NewClientCreds("")
+	if err != nil {
+		t.Fatalf("NewClientCreds(\"\") failed: %v", err)
+	}
+	if got, want := c.Info().ServerName, ""; got != want {
+		t.Errorf("c.Info().ServerName = %v, want %v", got, want)
+	}
+	if err := c.OverrideServerName(wantServerName); err != nil {
+		t.Fatalf("c.OverrideServerName(%v) failed: %v", wantServerName, err)
+	}
+	if got, want := c.Info().ServerName, wantServerName; got != want {
```

Comment:
nit: Can we add one more check to ensure c.serverName is equal to wantServerName?